File: MiniDOMOffice/CRM/Statistiche/ContattiInAttesa/ContattoInAttesaInfo.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using static minidom.Sistema;

namespace minidom
{
    public partial class CustomerCalls
    {

        /// <summary>
        /// Informazioni su una persona in attesa nel CRM
        /// </summary>
        [Serializable]
        public class ContattoInAttesaInfo 
            : Anagrafica.CPersonaInfo
        {

            /// <summary>
            /// Contatto in attesa (telefonata, visita, fax, ecc)
            /// </summary>
            public CContattoUtente Contatto;

            /// <summary>
            /// Costruttore
            /// </summary>
            public ContattoInAttesaInfo()
            {
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="contatto"></param>
            public ContattoInAttesaInfo(CContattoUtente contatto) : base()
            {
                if (contatto is null)
                    throw new ArgumentNullException("contatto");
                if (contatto.Persona is object)
                    Parse(contatto.Persona);
                Contatto = contatto;
            }

            /// <summary>
            /// Serializzazione xml
            /// </summary>
            /// <param name="writer"></param>
            protected override void XMLSerialize(XMLWriter writer)
            {
                base.XMLSerialize(writer);
                writer.WriteTag("Contatto", Contatto);
            }

            /// <summary>
            /// Deserializzazione xml
            /// </summary>
            /// <param name="fieldName"></param>
            /// <param name="fieldValue"></param>
            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "Contatto":
                        {
                            Contatto = (CContattoUtente)fieldValue;
                            break;
                        }

                    default:
                        {
                            base.SetFieldInternal(fieldName, fieldValue);
                            break;
                        }
                }
            }
        }
    }
}

File: MiniDOMOffice/CRM/Statistiche/ContattoInAttesaInfo.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Imports minidom.Anagrafica



Partial Public Class CustomerCalls

     <Serializable> _
    Public Class ContattoInAttesaInfo
        Inherits CPersonaInfo

        Public Contatto As CContattoUtente

        Public Sub New()
        End Sub

        Public Sub New(ByVal contatto As CContattoUtente)
            MyBase.New()
            If (contatto Is Nothing) Then Throw New ArgumentNullException("contatto")
            If (contatto.Persona IsNot Nothing) Then Me.Parse(contatto.Persona)
            Me.Contatto = contatto
        End Sub

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            MyBase.XMLSerialize(writer)
            writer.WriteTag("Contatto", Me.Contatto)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "Contatto" : Me.Contatto = fieldValue
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select
        End Sub
    End Class



End Class